wallet/src/program_facades/mod.rs

```diff
@@ -1,6 +1,142 @@
 //! This module contains [`WalletCore`](crate::WalletCore) facades for interacting with various
 //! on-chain programs.
 
+use nssa::{AccountId, ProgramId};
+use nssa_core::program::PdaSeed;
+use serde::{Serialize, ser::SerializeSeq};
+
+pub mod amm;
 pub mod native_token_transfer;
 pub mod pinata;
 pub mod token;
+
+fn compute_pool_pda(
+    amm_program_id: ProgramId,
+    definition_token_a_id: AccountId,
+    definition_token_b_id: AccountId,
+) -> AccountId {
+    AccountId::from((
+        &amm_program_id,
+        &compute_pool_pda_seed(definition_token_a_id, definition_token_b_id),
+    ))
+}
+
+fn compute_pool_pda_seed(
```

Comment:
I think in future it will be cool to have something like `amm_core` crate library to import into amm guest and wallet code to share this kind of functions.

Not related to this PR actually, just to mention.